templates/puppet.conf.erb

```diff
@@ -47,3 +47,13 @@
 
     syslogfacility = <%= @syslogfacility %>
 <% end -%>
+
+    
+<% if scope.lookupvar("puppet::server_directory_environments") -%>
+    environmentpath  = <%= scope.lookupvar("puppet::server_envs_dir") %>
+    basemodulepath   = <%= [scope.lookupvar("puppet::server_common_modules_path")].flatten.join(':') %>
+	<% elsif scope.lookupvar("puppet::server_git_repo") ||
+	        scope.lookupvar("puppet::server_dynamic_environments") -%>
+    manifest       = <%= scope.lookupvar("puppet::server_envs_dir") %>/$environment/manifests/site.pp
```

Comment:
I believe that ```[master]``` is an environment while this is the generic case since it refers to ```$environment```. That means it should be at least in ```[server]``` as it is now. That way you have the directory environments behaviour even though that was never intentionally supported.